tests/server/test_server_validation.py

```diff
@@ -51,3 +51,31 @@ def test_mongo_backend_package_used():
         raise Exception(
             f"The environment variable OPTIMADE_CI_FORCE_MONGO cannot be parsed as an int."
         )
+
+
+class AsTypeTestsWithValidator(SetClient, unittest.TestCase):
+
+    server = "regular"
+
+    def test_as_type_with_validator(self):
+
+        test_urls = {
+            "http://example.org/v0/structures": "structures",
+            "http://example.org/v0/structures/mpf_1": "structure",
+            "http://example.org/v0/references": "references",
+            "http://example.org/v0/references/dijkstra1968": "reference",
+            "http://example.org/v0/info": "info",
+            "http://example.org/v0/links": "links",
+        }
+        with unittest.mock.patch(
+            "requests.get", unittest.mock.Mock(side_effect=self.client.get)
```

Comment:
Huh. Didn't know about this., cool Is there a cleaner way to do this?

Reply:
You can do it slightly cleaner with pytest's monkeypatch, but that doesn't work nicely with `unittest.TestCase`, and we're relying on the client's being set with the `SetClient` metaclass which I didn't want to fiddle with, so I think this is the cleanest way without more changes to our setup.

Reply:
I think we should move to completely use `pytest` actually. And have indeed done this already for my own implementation (that basically copies these tests).

I will make a PR for this at some point.

Reply:
I'd agree, happy to help :)